### ISSUE TITLE
Static Extension failing in nightly builds (Flixel CI error)

### ISSUE THREAD

**Issue by Geokureli** [HaxeFoundation/haxe#11353]
Flixel CI has been failing nightly builds as of late: 
> \> haxelib run openfl build coverage html5 -Dcoverage2
> [/flixel/FlxSprite.hx:999](https://github.com/HaxeFlixel/flixel/blob/dev/flixel/FlxSprite.hx#L999): characters 19-33 : openfl.geom.ColorTransform has no field setMultipliers (Suggestion: redMultiplier)
> [/flixel/FlxSprite.hx:1001](https://github.com/HaxeFlixel/flixel/blob/dev/flixel/FlxSprite.hx#1001): characters 19-33 : openfl.geom.ColorTransform has no field setMultipliers (Suggestion: redMultiplier)

the issue is that the static extender class, [FlxColorTransformUtil](https://github.com/HaxeFlixel/flixel/blob/dev/flixel/util/FlxColorTransformUtil.hx#L7) seems to be ignored, despite being [used](https://github.com/HaxeFlixel/flixel/blob/dev/flixel/FlxSprite.hx#L26)

Note this only happens in the coverage test with coverage2 enabled which is [defined here](https://github.com/HaxeFlixel/flixel/blob/dev/tests/coverage/Project.xml#L53), it simply enables debug mode

It's hard to create a more simplified repro, but if there's anything else you need, I will try

**Comment by kLabz** [comment]
Sounds like a problem with the coverage macro. Where is it from?

**Comment by Simn** [comment]
Would be good to know which change broke this. It's probably module resolution, but static extensions shouldn't really be affected by it much, so I'm a bit curious.

**Comment by kLabz** [comment]
I suspect the error position is not the actual position, but a macro using that position and using some "deprecated" module resolution

**Comment by Geokureli** [comment]
I see this in BOTH the [earliest failing](https://github.com/HaxeFlixel/flixel/actions/runs/6662850278) and [latest succeeding](https://github.com/HaxeFlixel/flixel/actions/runs/6649407140) CI jobs:
> Resolved to nightly build 53e7a9f. Downloading ...

Earlier jobs have a different value. I assumed this would be some commit hash but the fact that it appears in both failing and succeeding jobs makes me think that perhaps this started happening when I changed a secondary flixel repo, like flixel-ui or flixel-addons, or maybe a change to lime or openfl.

**Comment by Geokureli** [comment]
> Sounds like a problem with the coverage macro. Where is it from?

coverage "macro"? I don't understand, I link the [coverage test's project.xml](https://github.com/HaxeFlixel/flixel/blob/dev/tests/coverage/Project.xml#L53) in the original post, if that helps

**Comment by Gama11** [comment]
That project [uses](https://github.com/HaxeFlixel/flixel/blob/c0a6f1c7fc7fc155641853968127a113a5b6b729/tests/coverage/Project.xml#L34C2-L34C87) `--macro include('flixel', true, ['flixel.system.macros'])`  to make sure everything is compiled.

**Comment by Geokureli** [comment]
> That project [uses](https://github.com/HaxeFlixel/flixel/blob/c0a6f1c7fc7fc155641853968127a113a5b6b729/tests/coverage/Project.xml#L34C2-L34C87) --macro include('flixel', true, ['flixel.system.macros']) to make sure everything is compiled.

@Gama11 I don't understand the implication of your post. Is there anything I need to do in order to fix Flixel CI or help fix this issue?

**Comment by Gama11** [comment]
That was more directed towards @kLabz' "what macro" question.

**Comment by kLabz** [comment]
Yeah I expected some macro doing "real coverage" instead of an "import all" thing.
Could be a typing order issue?

**Comment by Simn** [comment]
The `-Dcoverage1` version does not reproduce this for me:

```
C:\git\flixel\tests>haxelib run openfl build coverage neko -Dcoverage1
C:/git/spinehaxe/spinehaxe/atlas/Format.hx:33: characters 1-7 : Warning : (WDeprecatedEnumAbstract) `@:enum abstract` is deprecated in favor of `enum abstract`
C:/git/spinehaxe/spinehaxe/attachments/AttachmentType.hx:28: characters 1-7 : Warning : (WDeprecatedEnumAbstract) `@:enum abstract` is deprecated in favor of `enum abstract`
C:/git/spinehaxe/spinehaxe/animation/EventType.hx:33: characters 1-7 : Warning : (WDeprecatedEnumAbstract) `@:enum abstract` is deprecated in favor of `enum abstract`
C:/git/spinehaxe/spinehaxe/TransformMode.hx:33: characters 1-7 : Warning : (WDeprecatedEnumAbstract) `@:enum abstract` is deprecated in favor of `enum abstract`
C:/git/spinehaxe/spinehaxe/PositionMode.hx:33: characters 1-7 : Warning : (WDeprecatedEnumAbstract) `@:enum abstract` is deprecated in favor of `enum abstract`
C:/git/spinehaxe/spinehaxe/SpacingMode.hx:33: characters 1-7 : Warning : (WDeprecatedEnumAbstract) `@:enum abstract` is deprecated in favor of `enum abstract`
C:/git/spinehaxe/spinehaxe/RotateMode.hx:33: characters 1-7 : Warning : (WDeprecatedEnumAbstract) `@:enum abstract` is deprecated in favor of `enum abstract`
C:/git/spinehaxe/spinehaxe/BlendMode.hx:33: characters 1-7 : Warning : (WDeprecatedEnumAbstract) `@:enum abstract` is deprecated in favor of `enum abstract`
C:/git/spinehaxe/spinehaxe/atlas/TextureFilter.hx:33: characters 1-7 : Warning : (WDeprecatedEnumAbstract) `@:enum abstract` is deprecated in favor of `enum abstract`
C:/git/spinehaxe/spinehaxe/atlas/TextureWrap.hx:33: characters 1-7 : Warning : (WDeprecatedEnumAbstract) `@:enum abstract` is deprecated in favor of `enum abstract`
C:/git/spinehaxe/spinehaxe/animation/TimelineType.hx:33: characters 1-7 : Warning : (WDeprecatedEnumAbstract) `@:enum abstract` is deprecated in favor of `enum abstract`
C:/git/flixel-addons/flixel/addons/nape/FlxNapeSpace.hx:67: characters 3-39 : Warning : (WDeprecated) FlxG.plugins.add is deprecated, use `addIfUniqueType` or `addPlugin`, instead.
C:/git/flixel-addons/flixel/addons/nape/FlxNapeSpace.hx:67: characters 3-39 : ... Note: In a later version `add` will be changed to behave like `addPlugin`
C:\git\haxe\std/haxe/macro/Compiler.hx:396: characters 6-17 : Warning : (WDeprecated) Usage of this field is deprecated
C:\git\haxe\std/haxe/macro/Compiler.hx:408: characters 6-17 : Warning : (WDeprecated) Usage of this field is deprecated
C:\git\haxe\std/haxe/macro/Compiler.hx:423: characters 6-18 : Warning : (WDeprecated) Usage of this field is deprecated

C:\git\flixel\tests>echo %ERRORLEVEL%
0
```

With `-Dcoverage2` I do see it though:

```
C:/git/flixel/flixel/FlxSprite.hx:998: characters 19-33 : openfl.geom.ColorTransform has no field setMultipliers (Suggestion: redMultiplier
```

I don't know why that's the case and what it means.

**Comment by Simn** [comment]
And yes I confirm that this was caused by #11168.

**Comment by Simn** [comment]
Fixed by #11388.
